src/main/services/user_account/ResetApiCredsService.kt

```diff
@@ -1,2 +1,34 @@
 package main.services.user_account
 
+import kotlinserverless.framework.services.SOAResult
+import kotlinserverless.framework.services.SOAResultType
+import kotlinserverless.framework.services.SOAServiceInterface
+import main.daos.*
+import org.jetbrains.exposed.sql.transactions.transaction
+import java.lang.Exception
+
+/**
+ * This service will be used to reset a user account's ApiCreds
+ */
+object ResetApiCredsService: SOAServiceInterface<ApiCred> {
+    override fun execute(caller: Int?) : SOAResult<ApiCred> {
+        val apiCredResult = GenerateApiCredsService.execute()
+        if(apiCredResult.result != SOAResultType.SUCCESS)
+            return SOAResult(apiCredResult.result, apiCredResult.message, null)
+        val apiCredNamespace: ApiCredNamespace = apiCredResult.data!!
+        val apiCreds = ApiCred.new {
+            apiKey = apiCredNamespace.apiKey
+            secretKey = apiCredNamespace.secretKey
+        }
+
+        return try {
+            val userAccount = UserAccount.findById(caller!!)!!
+            transaction {
+                userAccount.apiCreds= apiCreds
+            }
+            SOAResult(SOAResultType.SUCCESS, null, userAccount.apiCreds)
```

Comment:
probably need to generate a new transaction -- look at GenerateUserAccountService for an example generating a new transaction -- the transaction will need to create a new one called "UPDATE" -- and it will be updating the useraccount object type